message_feed: Add channel settings link and remove Subscribe/Unsubscribe buttons.

reharsh opened the pull request:
This PR removes the Subscribe and Unsubscribe buttons from the message feed and adds links to either view channel settings (if not subscribed) or manage channel settings (if subscribed) in the bottom divider.

Fixes: #32125 

<details>
<summary>UI Changes</summary>

| Before | After |
| --- | --- |
| ![Before Image](https://github.com/user-attachments/assets/d2cdba99-9fd4-4def-93dc-366b3db8d9c0) | ![After Image](https://github.com/user-attachments/assets/3c62af51-115e-45ae-a3a8-26e051fad114) |
| ![Before Image](https://github.com/user-attachments/assets/b8855bfe-0b22-4c03-a2d0-34abacc44120) | ![After Image](https://github.com/user-attachments/assets/40f706cb-4adb-460d-a124-28a26b3e77e3) |
</details>

#### Demo Video

https://github.com/user-attachments/assets/208b55a4-9491-4388-9425-49a7c506232b

<details>

<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

reharsh commented:
Hey @alya! this PR is ready for review, Please take a look when you have a moment.

timabbott commented on web/src/message_list.ts:
```diff
@@ -430,9 +430,7 @@ export class MessageList {
-        const can_toggle_subscription =
-            sub !== undefined && stream_data.can_toggle_subscription(sub);
-        if (sub === undefined || sub.is_archived) {
+        if (sub === undefined) {
```
Removing the `sub.is_archived` check here is a bug.

timabbott commented on web/tests/message_list.test.js:
```diff
@@ -359,19 +359,20 @@ run_test("bookend", ({override}) => {
 
     override(stream_data, "is_subscribed", () => is_subscribed);
     override(stream_data, "get_sub_by_id", () => ({invite_only, name: "IceCream"}));
-    override(stream_data, "can_toggle_subscription", () => true);
```
These changes should be squashed into the commit that makes the test fail.

Check out our [GitHub guide](https://zulip.readthedocs.io/en/latest/git-guide.html) and [commit guidelines](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html) for more details.

reharsh replied:
done, now each commit passes all tests individually. I’ll keep this in mind for future PRs.

prakhar1144 commented:
@reharsh 

See: https://zulip.readthedocs.io/en/latest/contributing/review-process.html#how-to-help-move-the-review-process-forward

reharsh commented:
> @reharsh
> 
> See: https://zulip.readthedocs.io/en/latest/contributing/review-process.html#how-to-help-move-the-review-process-forward

Sure, I'll ping Tim for a review, as I believe there’s nothing more on my end to address. Thanks for pointing me to the review process guidelines!

reharsh commented:
@timabbott, I've made the following updates:

- Fixed a bug with `sub.is_archived` that was mistakenly removed.
- Squashed commits that add `channels-settings-link` and update the relevant test files into one.

Please take a look when you have a chance. Thank you!

timabbott commented:
@reharsh I can fix it this time, but can you line-wrap your commit messages? They don't look good in `git show`. 

Merged after those changes, thanks @reharsh!
